### ISSUE TITLE
Error in MOI function implementation?

### ISSUE THREAD

**Issue by kibaekkim** [JuliaOpt/AmplNLWriter.jl#95]
I would like to try an AMPL-enabled `solver` with the following snippet, which was run on Mac OSC with Julia 1.12.

```
using PowerModels
using AmplNLWriter

instance = "/Users/kibaekkim/.julia/packages/PowerModels/w29A2/test/data/matpower/case30.m"

run_opf(instance,
    ACRPowerModel, 
    with_optimizer(AmplNLWriter.Optimizer, solver)
)
```

With that, I got the following errors:
```
MethodError: supports_constraint(::AmplNLWriter.InnerModel{Float64}, ::Type{MathOptInterface.SingleVariable}, ::Type{MathOptInterface.GreaterThan{Float64}}) is ambiguous. Candidates:
  supports_constraint(model::AmplNLWriter.InnerModel{#76#T}, ::Type{#s21} where #s21<:Union{MathOptInterface.SingleVariable, MathOptInterface.ScalarAffineFunction{#76#T}, MathOptInterface.ScalarQuadraticFunction{#76#T}}, ::Type{#s20} where #s20<:Union{MathOptInterface.Integer, MathOptInterface.ZeroOne, MathOptInterface.EqualTo{#76#T}, MathOptInterface.GreaterThan{#76#T}, MathOptInterface.Interval{#76#T}, MathOptInterface.LessThan{#76#T}}) where #76#T in AmplNLWriter at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/model.jl:1020
  supports_constraint(::MathOptInterface.Utilities.AbstractModel{T}, ::Type{MathOptInterface.SingleVariable}, ::Type{#s118} where #s118<:Union{MathOptInterface.Integer, MathOptInterface.ZeroOne, MathOptInterface.EqualTo{T}, MathOptInterface.GreaterThan{T}, MathOptInterface.Interval{T}, MathOptInterface.LessThan{T}, MathOptInterface.Semicontinuous{T}, MathOptInterface.Semiinteger{T}}) where T in MathOptInterface.Utilities at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/model.jl:489
Possible fix, define
  supports_constraint(::AmplNLWriter.InnerModel{T}, ::Type{MathOptInterface.SingleVariable}, ::Type{#s118} where #s118<:Union{MathOptInterface.Integer, MathOptInterface.ZeroOne, MathOptInterface.EqualTo{T}, MathOptInterface.GreaterThan{T}, MathOptInterface.Interval{T}, MathOptInterface.LessThan{T}, MathOptInterface.Semicontinuous{T}, MathOptInterface.Semiinteger{T}})

Stacktrace:
 [1] supports_constraint(::MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}, ::Type{MathOptInterface.SingleVariable}, ::Type{MathOptInterface.GreaterThan{Float64}}) at /Users/kibaekkim/.julia/packages/AmplNLWriter/V1gW5/src/MOI_wrapper.jl:34
 [2] is_bridged(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}}, ::Type{MathOptInterface.GreaterThan{Float64}}) at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Bridges/lazy_bridge_optimizer.jl:267
 [3] add_constrained_variable(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}}, ::MathOptInterface.GreaterThan{Float64}) at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Bridges/bridge_optimizer.jl:1174
 [4] copy_single_variable(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::MathOptInterface.Utilities.IndexMap, ::Type{MathOptInterface.GreaterThan{Float64}}) at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/copy.jl:203
 [5] (::getfield(MathOptInterface.Utilities, Symbol("##121#129")){MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}},MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}},MathOptInterface.Utilities.IndexMap})(::Type) at ./none:0
 [6] collect(::Base.Generator{Array{DataType,1},getfield(MathOptInterface.Utilities, Symbol("##121#129")){MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}},MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}},MathOptInterface.Utilities.IndexMap}}) at ./generator.jl:47
 [7] default_copy_to(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}, ::Bool) at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/copy.jl:325
 [8] #automatic_copy_to#103 at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/copy.jl:15 [inlined]
 [9] #automatic_copy_to at ./none:0 [inlined]
 [10] #copy_to#3 at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Bridges/bridge_optimizer.jl:268 [inlined]
 [11] (::getfield(MathOptInterface, Symbol("#kw##copy_to")))(::NamedTuple{(:copy_names,),Tuple{Bool}}, ::typeof(MathOptInterface.copy_to), ::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.UniversalFallback{AmplNLWriter.InnerModel{Float64}}}, ::MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}) at ./none:0
 [12] attach_optimizer(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{MathOptInterface.Utilities.Model{Float64}}}) at /Users/kibaekkim/.julia/packages/MathOptInterface/izyVf/src/Utilities/cachingoptimizer.jl:149
 [13] attach_optimizer(::JuMP.Model) at /Users/kibaekkim/.julia/packages/JuMP/MsUSY/src/optimizer_interface.jl:32
 [14] #optimize!#78(::Bool, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(JuMP.optimize!), ::JuMP.Model, ::JuMP.OptimizerFactory) at /Users/kibaekkim/.julia/packages/JuMP/MsUSY/src/optimizer_interface.jl:125
 [15] optimize! at /Users/kibaekkim/.julia/packages/JuMP/MsUSY/src/optimizer_interface.jl:111 [inlined]
 [16] macro expansion at ./util.jl:289 [inlined]
 [17] #optimize_model!#199(::JuMP.OptimizerFactory, ::Function, ::typeof(optimize_model!), ::ACRPowerModel) at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/core/base.jl:207
 [18] (::getfield(PowerModels, Symbol("#kw##optimize_model!")))(::NamedTuple{(:optimizer, :solution_builder),Tuple{JuMP.OptimizerFactory,typeof(solution_opf!)}}, ::typeof(optimize_model!), ::ACRPowerModel) at ./none:0
 [19] #run_model#196(::Array{Any,1}, ::Function, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(run_model), ::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory, ::Function) at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/core/base.jl:151
 [20] run_model(::Dict{String,Any}, ::Type, ::JuMP.OptimizerFactory, ::Function) at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/core/base.jl:147
 [21] #run_model#195(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(run_model), ::String, ::Type, ::JuMP.OptimizerFactory, ::Function) at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/core/base.jl:141
 [22] run_model at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/core/base.jl:140 [inlined]
 [23] #run_opf#801 at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/prob/opf.jl:13 [inlined]
 [24] run_opf(::String, ::Type, ::JuMP.OptimizerFactory) at /Users/kibaekkim/.julia/packages/PowerModels/w29A2/src/prob/opf.jl:13
 [25] top-level scope at In[23]:6
```

Does anyone have a good guess?

**Comment by odow** [comment]
Looks like we need to implement a method like
```Julia
function MOI.supports_constraint(
    ::InnerModel, 
    ::Type{MOI.SingleVariable},
    ::Type{Union{
        MOI.GreaterThan, MOI.LessThan, MOI.EqualTo, MOI.Interval, 
        MOI.ZeroOne, MOI.Integer
    }}
) where T
    return true
end
```

If you make it `AmplNLWriter.InnerModel`, you should be able to add the function to your script to check it resolves the issue.

**Comment by ccoffrin** [comment]
Repeat of https://github.com/JuliaOpt/AmplNLWriter.jl/issues/49?

**Comment by odow** [comment]
Not a duplicate. But resolved by #99 anyhow.
